entity/c/c_secure_comm.h

```diff
@@ -0,0 +1,137 @@
+// @return total buffer with session key id and encrypted message
+unsigned char *parse_handshake_1(session_key_t *s_key, unsigned char *entity_nonce, unsigned int *ret_length);
+
+// Check the nonce obtained in decryption with own nonce and
+// make the encrypted message with other entity's nonce.
+// @param data_buf input data buffer
+// @param data_buf_length length of data buffer
+// @param entity_nonce own nonce
+// @param s_key session key struct
+// @param ret_length length of return buffer
+// @return buffer with encrypted message
+unsigned char *check_handshake_2_send_handshake_3(unsigned char *data_buf, unsigned int data_buf_length, unsigned char *entity_nonce, session_key_t *s_key, unsigned int *ret_length);
+
+// Decrypts message, reads seq_num, checks validity, and prints message
+// Print the received message and sequence number after check validity of session key.
+// @param data input data buffer
+// @param data_length length of data buffer
+// @param s_key session key struct
+void print_recevied_message(unsigned char *data, unsigned int data_length, session_key_t *s_key);
+
+// Check the validity of session key by calculating relative time and absolute time.
+// @param seq_n sequence number of received message
+// @param rel_validity relative validity time of session key
+// @param abs_validity absolute validity time of session key
+// @param st_time time of first use of session key
+// @return 1 or 0 depending on validity
+int check_validity(int seq_n, unsigned char *rel_validity, unsigned char *abs_validity, long int *st_time);
+
+// Check if entity has session key and if not, request the session key to Auth.
+// @param config config struct for the entity information
+// @param target_key_id id of session key
+// @return session key struct according to key id
+session_key_t *send_session_key_request_check_protocol(config_t *config, unsigned char *target_key_id);
+
+// Request the session key to Auth according to session key id via TCP connection
+// @param config_info config struct for the entity information
+// @return session_key_t struct according to key id
+session_key_t *send_session_key_req_via_TCP(config_t *config);
+
+// Request the session key to Auth according to session key id via UDP connection.
+// @param
+// @return session key struct according to key id
+session_key_t *send_session_key_req_via_UDP();
+
+// Check the nonce obtained in decryption with own nonce and
+// make the encrypted message with other entity's nonce.
+// @param received_buf received buffer
+// @param received_buf_length length of received buffer
+// @param server_nonce own nonce
+// @param s_key session key struct
+// @param ret_length length of return buffer
+// @return buffer with encrypted message
+unsigned char *check_handshake1_send_handshake2(unsigned char *received_buf, unsigned int received_buf_length, unsigned char *server_nonce, session_key_t *s_key, unsigned int *ret_length);
+
+#endif // C_SECURE_COMM_H
```

Comment:
I recommend adding a new line for each file.

Reply:
What does this mean?

Reply:
Each file should be terminated with a new line.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Done